wgpu/build.rs

```diff
@@ -0,0 +1,14 @@
+fn main() {
+    cfg_aliases::cfg_aliases! {
+        native: { not(target_arch = "wasm32") },
+        webgl: { all(target_arch = "wasm32", not(target_os = "emscripten"), feature = "webgl") },
+        webgpu: { all(target_arch = "wasm32", not(target_os = "emscripten"), not(feature = "webgl")) },
+        emscripten: { all(target_arch = "wasm32", target_os = "emscripten") },
+        send_sync: { any(
+            not(target_arch = "wasm32"),
+            all(feature = "fragile-send-sync-non-atomic-wasm", not(target_feature = "atomics"))
+        ) },
+        dx12: { all(target_os = "windows", feature = "dx12") },
+        metal: { all(any(target_os = "ios", target_os = "macos"), feature = "metal") }
```

Comment:
I would recommend to give those nicer names as they are visible in the documentation:
![image](https://github.com/gfx-rs/wgpu/assets/1645124/9aff6b37-c9b2-47a0-9d8f-3d898b91b568)
We could use "WebGPU", "DX12" and "Metal". This has the drawback that seeing capitalized letters in `cfg` guards is a bit odd.

Reply:
hmm yeah I'm a bit torn, but as you say capitalized cfgs are very unusual, so I'd stick with that regardless. This is still _sooo_ much better anyways 😄